pkg/credentialprovider/config.go

```diff
@@ -30,6 +30,13 @@ import (
 	"github.com/golang/glog"
 )
 
+// DockerConfigJson represents ~/.docker/config.json file info
+// see https://github.com/docker/docker/pull/12009
+type DockerConfigJson struct {
+	Auths       DockerConfig      `json:"auths"`
+	HttpHeaders map[string]string `json:"HttpHeaders,omitempty"`
```

Comment:
This struct `DockerConfigJson` is used to store content of `~/.docker/config.json`, it is the same structure as [ConfigFile](https://github.com/docker/docker/blob/master/cliconfig/config.go#L57) struct in `docker/cliconfig`.
```go
// ConfigFile ~/.docker/config.json file info
type ConfigFile struct {
	AuthConfigs map[string]AuthConfig `json:"auths"`
	HTTPHeaders map[string]string     `json:"HttpHeaders,omitempty"`
	PsFormat    string                `json:"psFormat,omitempty"`
	filename    string                // Note: not serialized - for internal use only
}
```
In the case we don't use `HTTPHeaders` and `PsFormat`, then we do not need that struct. And as in your comment below, a single method `func DockerConfigFrom(bytes []bytes) (DockerConfig, error)` which parse both new and old type of secrets will make code easier to maintain.